test/e2e/resources/mockComponentPartial.brs

```diff
@@ -0,0 +1,42 @@
+sub main()
+    print formatJson({}) ' => {real: 'json'}
+    print get() ' => Status: 500
+    print post() ' => Status: 200
+    print isValid() ' => false
+
+    _brs_.mockComponentPartial({
+        formatJson: _brs_.mockFunction("formatJson", function(json as object, flags=0 as integer) as dynamic
```

Comment:
So I think that the reason this test is passing is because we're calling `_brs_.mockFunction` here. This is equivalent to making that call without also calling `_brs_.mockComponentPartial`:
```brightscript
print isValid() ' => false

_brs_.mockFunction("formatJson", function(json as object, flags=0 as integer) as dynamic
    return "{fake:'json'}"
end function)
...
```
The reason it works is because `_brs_.mockFunction` [mocks any non-local function](https://github.com/sjbarag/brs/blob/master/src/interpreter/Environment.ts#L163-L196), so it mocks the functions you declared at the bottom of this file.

I think the way that we'd probably want to test this is something like:
```
_brs_.mockComponentPartial({
        formatJson: function(json as object, flags=0 as integer) as dynamic
            return "{fake:'json'}"
        end function)
})
```
The difference here is that we're passing in an associative array, where the values are the mock functions themselves, rather than calling `_brs_.mockFunction`.